### PR TITLE
Fix unstored audio

### PR DESCRIPTION
**Pull request by lherman-cs**
* Add unit test to check internal buffer integrity
* Fix unstored audio in internal buffer

### CODE REVIEW

File: pkg/wave/buffer_test.go [hunk 1]
```diff
@@ -12,8 +12,9 @@ func TestBufferStoreCopyAndLoad(t *testing.T) {
 		SamplingRate: 48000,
 	}
 	testCases := map[string]struct {
-		New    func() EditableAudio
-		Update func(EditableAudio)
+		New      func() EditableAudio
+		Update   func(EditableAudio)
+		Validate func(Audio, Audio) bool
```
Comment (at-wat) [inline pkg/wave/buffer_test.go:17]:
```suggestion
		Validate func(*testing.T, Audio, Audio) bool
```
would be better to make it easy to find why the validation is failed.

Comment (lherman-cs) [reply pkg/wave/buffer_test.go:17]: I see. In that case, I'll just return error then instead, and let the higher level to print that.

Comment (at-wat) [reply pkg/wave/buffer_test.go:17]: `t.Error` outputs the filename and line number where it errored. Calling `t.Error` in Validate func makes it easy to find the failing line.

Comment (lherman-cs) [reply pkg/wave/buffer_test.go]: Oh! I didn't know that. I'll do that then!

Comment (lherman-cs) [reply pkg/wave/buffer_test.go]: Updated.

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**